### PR TITLE
Develop merge main to develop

### PR DESCRIPTION
**Pull request by HarryLiIsMe**
* **make sure that you have executed all the following process and no errors occur**
  - [ ] make fmt
  - [ ] make lint
  - [ ] make test

* **The major changes of this PR**
1. Merge fixes/changes in main branch into develop branch

The Manual Test result is passing: https://jenkins.dev.findora.org/view/integration%20test/job/Manual-Test/60/console

* **The major impacts of this PR**
  - [ ] Impact WASM?
  - [ ] Impact Web3 API?
  - [ ] Impact mainnet data compatibility?

* **Extra documentations**

### CODE REVIEW

**Comment by ktmlm** [comment]
a 'rebase' should be better than a 'reverse merge'

**Comment by HarryLiIsMe** [comment]
> a 'rebase' should be better than a 'reverse merge'
i have tried it, but because the version of the develop and the main is too large different will brings many new problems.

File: .github/workflows/Develop.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 name: Develop
 on:
+  push:
```
Comment (ws4charlie) [inline .github/workflows/Develop.yml:3]: github workflows are managed by Jimmy's team I think. We don't need to touch maybe.

File: .github/workflows/Main.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 name: Main
 on:
+  push:
```
Comment (ws4charlie) [inline .github/workflows/Main.yml:3]: Maintained by Jimmy's team.

**Review by ws4charlie — APPROVED** [review]

File: .github/workflows/Main.yml (hunk 2 above)
Comment (HarryLiIsMe) [reply .github/workflows/Main.yml]: ok, i have done it

File: .github/workflows/Develop.yml (hunk 1 above)
Comment (HarryLiIsMe) [reply .github/workflows/Develop.yml]: ok, i have done it

**Pull request merged**